react/react1/week3/todo-list/src/components/AddForm.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react";
+import uniqid from "uniqid";
+
+export const AddForm = ({ todos, setTodos }) => {
+  //Defining states for the input fields
+  const [inputDescription, setDescription] = useState("");
+  const [inputDeadline, setDeadline] = useState("");
+
+  const addHandler = () => {
+    const newTodo = {
+      id: uniqid(),
+      description: inputDescription,
+      deadline: inputDeadline,
+    };
+
+    //We should do a bit of validation first not to get an empty todo
+    if (newTodo.description !== "" && newTodo.deadline !== null) {
+      const today = new Date();
+      const deadline = new Date(newTodo.deadline);
+      if (deadline < today) {
+        alert(
+          "Oops, looks like the deadline has already passed. Please change it to either today or later!"
+        );
+      } else {
+        setTodos(todos.concat(newTodo));
+      }
+    } else {
+      alert("You should assign both a description and a deadline");
+    }
+
+    console.log(newTodo.id);
+
+    // Reset input values
+    setDescription("");
+    document.getElementById("descriptionInput").value = "";
```

Comment:
Would bind the value of the input to a state variable then simply setting the state to be an empty string would do the same thing but be a lot easier to understand.